libraries/plugins/es_objects/es_objects.cpp

```diff
@@ -0,0 +1,301 @@
+{
+   return "Stores blockchain objects in ES database.";
+}
+
+void es_objects_plugin::plugin_set_program_options(
+   boost::program_options::options_description& cli,
+   boost::program_options::options_description& cfg
+   )
+{
+   cli.add_options()
+         ("es-objects-elasticsearch-url", boost::program_options::value<std::string>(), "Elasticsearch node url")
+         ("es-objects-logs", boost::program_options::value<bool>(), "Log bulk events to database")
+         ("es-objects-bulk-replay", boost::program_options::value<uint32_t>(), "Number of bulk documents to index on replay(1000)")
+         ("es-objects-bulk-sync", boost::program_options::value<uint32_t>(), "Number of bulk documents to index on a syncronied chain(2)")
+         ("es-objects-proposals", boost::program_options::value<bool>(), "Store proposal objects")
+         ("es-objects-accounts", boost::program_options::value<bool>(), "Store account objects")
+         ("es-objects-assets", boost::program_options::value<bool>(), "Store asset objects")
+         ("es-objects-balances", boost::program_options::value<bool>(), "Store balances objects")
+         ("es-objects-limit-orders", boost::program_options::value<bool>(), "Store limit order objects")
+
+         ;
+   cfg.add(cli);
+}
+
+void es_objects_plugin::plugin_initialize(const boost::program_options::variables_map& options)
+{
+   database().new_objects.connect([&]( const vector<object_id_type>& ids, const flat_set<account_id_type>& impacted_accounts ){ my->updateDatabase(ids, 1); });
+   database().changed_objects.connect([&]( const vector<object_id_type>& ids, const flat_set<account_id_type>& impacted_accounts ){ my->updateDatabase(ids, 0); });
+
+   if (options.count("es-objects-elasticsearch-url")) {
+      my->_es_objects_elasticsearch_url = options["es-objects-elasticsearch-url"].as<std::string>();
+   }
+   if (options.count("es-objects-logs")) {
+      my->_es_objects_logs = options["es-objects-logs"].as<bool>();
+   }
+   if (options.count("es-objects-bulk-replay")) {
+      my->_es_objects_bulk_replay = options["es-objects-bulk-replay"].as<uint32_t>();
+   }
+   if (options.count("es-objects-bulk-sync")) {
+      my->_es_objects_bulk_sync = options["es-objects-bulk-sync"].as<uint32_t>();
+   }
+   if (options.count("es-objects-proposals")) {
+      my->_es_objects_proposals = options["es-objects-proposals"].as<bool>();
+   }
+   if (options.count("es-objects-accounts")) {
+      my->_es_objects_accounts = options["es-objects-accounts"].as<bool>();
+   }
+   if (options.count("es-objects-assets")) {
+      my->_es_objects_assets = options["es-objects-assets"].as<bool>();
+   }
+   if (options.count("es-objects-balances")) {
+      my->_es_objects_balances = options["es-objects-balances"].as<bool>();
+   }
+   if (options.count("es-objects-limit-orders")) {
+      my->_es_objects_limit_orders = options["es-objects-limit-orders"].as<bool>();
+   }
+}
+
+void es_objects_plugin::plugin_startup()
+{
```

Comment:
Can you add a log entry here so people have feedback about es plugin being enabled when starting the backend, please?

Reply:
added at https://github.com/bitshares/bitshares-core/pull/500/commits/109b19f79101a36b7033eee357d3da50751f4b35 thx.